### PR TITLE
feat: merge cw address list contract

### PR DESCRIPTION
**Pull request by WaDadidou**

### CODE REVIEW

File: rust/cw-contracts/cw-address-list/Makefile [hunk 1]
```diff
@@ -0,0 +1,54 @@
+ADMIN_TESTNET=testnet-adm
```
Comment (n0izn0iz) [inline rust/cw-contracts/cw-address-list/Makefile:1]: we should remove this makefile and use scripts that are common to all contracts IMO

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

File: rust/cw-contracts/cw-address-list/Makefile (hunk 1 above)
Comment (WaDadidou) [reply rust/cw-contracts/cw-address-list/Makefile]: Done

File: packages/networks/types.ts [hunk 2]
```diff
@@ -72,6 +72,8 @@ export type CosmosNetworkInfo = NetworkInfoBase & {
   daoVotingCw4CodeId?: number;
   cwAdminFactoryContractAddress?: string;
   coreDAOAddress?: string;
+  cwAddressListContractAddress?: string;
+  cwAddressListCodeId?: number;
```
Comment (n0izn0iz) [inline packages/networks/types.ts:76]: please add a NetworkFeature if you need this in this PR

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

File: packages/scripts/integration-testing/cwAddresslistTest.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { program } from "commander";
+import fs from "fs/promises";
+
+import { deployCwAddressList } from "./../network-setup/deployCwAddressList";
```
Comment (n0izn0iz) [inline packages/scripts/integration-testing/cwAddresslistTest.ts:4]: using this import will also trigger the `main` func in `deployCwAddressList`

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

File: packages/scripts/integration-testing/cwAddresslistTest.ts (hunk 3 above)
Comment (WaDadidou) [reply packages/scripts/integration-testing/cwAddresslistTest.ts:4]: Removed. 
We can test using deployCwAddressList with localnet

File: packages/networks/types.ts (hunk 2 above)
Comment (WaDadidou) [reply packages/networks/types.ts:76]: Done

File: packages/networks/features.ts [hunk 4]
```diff
@@ -17,8 +17,19 @@ export enum NetworkFeature {
...
+  CosmWasmAddressList = "CosmWasmAddressList",
 }
 
+// CosmWasm Address List
+
+const zodCCosmWasmAddressList = z.object({
+  type: z.literal(NetworkFeature.CosmWasmAddressList),
+  cwAddressListContractAddress: z.string(),
+  codeId: z.number(),
+});
```
Comment (n0izn0iz) [inline packages/networks/features.ts:29]: this feature should probably be called `NFTMarketplaceFeature`
there is no "address list feature", the address list is a tool that is used as the whitelist of the marketplace feature

we can move the other fields into it later

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

File: packages/networks/features.ts (hunk 4 above)
Comment (WaDadidou) [reply packages/networks/features.ts:29]: Wa sorry I'm dumb today

**Pull request merged**